app/controllers/mechanic_rides_controller.rb

```diff
@@ -0,0 +1,15 @@
+class MechanicRidesController < ApplicationController
+  def update
+    @mechanic = Mechanic.find(params[:id])
+    @mechanic_rides = @mechanic.mechanic_rides
+
+    @mechanic_rides.update(update_params)
+    redirect_to mechanic_path(@mechanic)
+  end
```

Comment:
When looking at US2, here is what we're being asked to do: `"When I go to a mechanic's show page, I see a form to ADD a ride to their workload with an ID of an existing ride"`.

I can see that you're trying to do that using an `update` but this isn't doing what you think it is. 

Let's break down what's going on here. You grab all of the mechanic's rides and store them in a variable `@mechanic_rides` which now contains a collection of `Ride` objects. And then you are calling `update` on that collection. The issue is, what is being updated? When calling on `update`, this usually updates one or more columns on an instance object, not on a collection. 

This also doesn't align with the functionality that we are trying to implement. Since `a mechanic can HAVE MANY rides` and `a ride can HAVE MANY mechanics`, when adding an additional ride to a mechanic, this is going to go on our joins table where we `CREATE` a new row linking them together.

Here is what the code could look like:

```ruby
def create
  mechanic = Mechanic.find(params[:id])
  ride = Ride.find(params[:ride_id])

  MechanicRide.create(mechanic: mechanic, ride: ride)
  redirect_to mechanic_path(mechanic)
end
```